viadot/sources/epicor.py

```diff
@@ -34,7 +201,8 @@ def __init__(
         DEFAULT_CREDENTIALS = local_config.get(config_key)
         credentials = credentials or DEFAULT_CREDENTIALS
 
-        if credentials is None:
+        required_credentials = ["host", "port", "username", "password"]
+        if any([cred_key not in credentials for cred_key in required_credentials]):
             raise CredentialError("Credentials not found.")
```

Comment:
Kind of misleading as it could be just one missing credential. You could also easily fetch and show it to the user:
```python
not_found = [c for c in required_credentials if c not in credentials]
f"Missing credential(s): '{not_found}'."
```